src/ordered-elements.js

```diff
@@ -19,9 +19,12 @@ export default class OrderedElements {
         }
     }
 
-    set(key /* : string */, value /* : any */) {
+    set(key /* : string */, value /* : any */, preserveOrder /* : boolean */) {
         if (!this.elements.hasOwnProperty(key)) {
             this.keyOrder.push(key);
+        } else if (!preserveOrder) {
+            const index = this.keyOrder.indexOf(key);
+            this.keyOrder.push(this.keyOrder.splice(index, 1)[0]);
```

Comment:
I think splitting this into two lines would be clearer
```
this.keyOrder.splice(index, 1);
this.keyOrder.push(key);
```

Reply:
Ah, good call.

Reply:
oops, argh, I meant to put this into a review. Other comments incoming...